diff --git a/ArduPlane/Plane.h b/ArduPlane/Plane.h
index 207b11935a..cdde1df7be 100644
--- a/ArduPlane/Plane.h
+++ b/ArduPlane/Plane.h
@@ -744,6 +744,7 @@ private:
 
         // current value of loiter radius in metres used by the controller
         float radius;
+
     } loiter;
 
     // Conditional command
@@ -1313,6 +1314,7 @@ private:
     int32_t tecs_target_alt_cm;
 
 public:
+
     void failsafe_check(void);
     bool is_landing() const override;
     bool is_taking_off() const override;
diff --git a/libraries/AP_OSD/AP_OSD.h b/libraries/AP_OSD/AP_OSD.h
index 847eb6d48e..e00b29d831 100644
--- a/libraries/AP_OSD/AP_OSD.h
+++ b/libraries/AP_OSD/AP_OSD.h
@@ -216,6 +216,7 @@ private:
 #endif
     AP_OSD_Setting hdop;
     AP_OSD_Setting waypoint;
+    AP_OSD_Setting loiter_radius;
     AP_OSD_Setting xtrack_error;
     AP_OSD_Setting dist{false,22,11};
     AP_OSD_Setting stat{false,0,0};
@@ -318,6 +319,7 @@ private:
 #endif
     void draw_hdop(uint8_t x, uint8_t y);
     void draw_waypoint(uint8_t x, uint8_t y);
+    void draw_loiter_radius(uint8_t x, uint8_t y);
     void draw_xtrack_error(uint8_t x, uint8_t y);
     void draw_dist(uint8_t x, uint8_t y);
     void draw_stat(uint8_t x, uint8_t y);
diff --git a/libraries/AP_OSD/AP_OSD_Screen.cpp b/libraries/AP_OSD/AP_OSD_Screen.cpp
index 97e3831b2d..6d7b8b1a66 100644
--- a/libraries/AP_OSD/AP_OSD_Screen.cpp
+++ b/libraries/AP_OSD/AP_OSD_Screen.cpp
@@ -43,6 +43,7 @@
 #include <AP_Terrain/AP_Terrain.h>
 #include <AP_RangeFinder/AP_RangeFinder.h>
 #include <AP_Vehicle/AP_Vehicle.h>
+#include "../../ArduPlane/Plane.h"
 #include <AP_RPM/AP_RPM.h>
 #include <AP_MSP/AP_MSP.h>
 #if APM_BUILD_TYPE(APM_BUILD_Rover)
@@ -1045,6 +1046,22 @@ const AP_Param::GroupInfo AP_OSD_Screen::var_info[] = {
 	AP_SUBGROUPINFO(rrpm, "RPM", 62, AP_OSD_Screen, AP_OSD_Setting),
 #endif
 
+    // @Param: LOITRAD_EN
+    // @DisplayName: LOITRAD_EN
+    // @Description: Displays loiter radius
+    // @Values: 0:Disabled,1:Enabled
+
+    // @Param: LOITRAD_X
+    // @DisplayName: LOITRAD_X
+    // @Description: Horizontal position on screen
+    // @Range: 0 59
+
+    // @Param: LOITRAD_Y
+    // @DisplayName: LOITRAD_Y
+    // @Description: Vertical position on screen
+    // @Range: 0 21
+    AP_SUBGROUPINFO(loiter_radius, "LOITRAD", 63, AP_OSD_Screen, AP_OSD_Setting),
+
     AP_GROUPEND
 };
 
@@ -2273,6 +2290,21 @@ void AP_OSD_Screen::draw_waypoint(uint8_t x, uint8_t y)
     draw_distance(x+4, y, osd->nav_info.wp_distance);
 }
 
+void AP_OSD_Screen::draw_loiter_radius(uint8_t x, uint8_t y)
+{
+    enum ap_var_type ptype;
+    AP_Int16 *loiter_rad =
+        (AP_Int16 *)AP_Param::find("WP_LOITER_RAD", &ptype);
+
+    int radius_ft = 0;
+
+    if (loiter_rad != nullptr) {
+        radius_ft = (int)((float)loiter_rad->get() * 3.28084f);
+    }
+
+    backend->write(x, y, false, "R:%d", radius_ft);
+}
+
 void AP_OSD_Screen::draw_xtrack_error(uint8_t x, uint8_t y)
 {
     backend->write(x, y, false, "%c", SYMBOL(SYM_XERR));
@@ -2578,6 +2610,7 @@ void AP_OSD_Screen::draw(void)
     DRAW_SETTING(rngf);
 #endif
     DRAW_SETTING(waypoint);
+    DRAW_SETTING(loiter_radius);
     DRAW_SETTING(xtrack_error);
     DRAW_SETTING(bat_volt);
     DRAW_SETTING(bat2_vlt);
